### PR TITLE
add Sendsmaily subscribe possibility

### PR DESCRIPTION
**Pull request by qwestern**

### CODE REVIEW

File: src/senders/Sendsmaily.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace nuffic\subscribe\senders;
+
+use nuffic\sendsmaily\ApiClient;
```
Comment (Nuffic) [inline src/senders/Sendsmaily.php:5]: where does ApiClient come from?

File: src/senders/Sendsmaily.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+
+    public $baseUrl;
+    public $apiUser;
+    public $apiPassword;
+
+    public function subscribe(Model $model)
+    {
+
+        $result = $this->addSubscriber($model->email);
```
Comment (Nuffic) [inline src/senders/Sendsmaily.php]: as sendsmaily also accepts params related to email, please make it possible to send those

File: src/senders/Sendsmaily.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace nuffic\subscribe\senders;
+
+use nuffic\sendsmaily\ApiClient;
+use Yii;
+use yii\base\Model;
+
+class Sendsmaily extends ApiClient
+{
+
```
Comment (Nuffic) [inline src/senders/Sendsmaily.php:11]: remove empty line

File: src/senders/Sendsmaily.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public $baseUrl;
+    public $apiUser;
+    public $apiPassword;
+
+    public function subscribe(Model $model)
+    {
+
```
Comment (Nuffic) [inline src/senders/Sendsmaily.php]: remove empty line

File: src/senders/Sendsmaily.php [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        $result = $this->addSubscriber($model->email);
+
+        if (strtoupper($result->message) == 'OK') {
+            return $model;
+        }
+
+        $model->addError('email', Yii::t('app', 'Something went wrong'));
+        return $model;
+    }
+}
```
Comment (Nuffic) [inline src/senders/Sendsmaily.php:28]: missing EOL on EOF

**Review by Nuffic — CHANGES_REQUESTED** [review]

File: src/senders/Sendsmaily.php (hunk 1 above)
Comment (qwestern) [reply src/senders/Sendsmaily.php:5]:
```
    "nuffic/sendsmaily": "dev-develop"
```

Comment (Nuffic) [reply src/senders/Sendsmaily.php:5]: now, yes :)

File: src/senders/Sendsmaily.php [hunk 6]
```diff
@@ -16,7 +16,7 @@ class Sendsmaily extends ApiClient
 
     public function subscribe(Model $model)
     {
-        $result = $this->addSubscriber($model->email, $this->params);
+        $result = $this->addSubscriber($model->email,(array) $this->params);
```
Comment (Nuffic) [inline src/senders/Sendsmaily.php]: this is casting not checking

**Pull request merged**